#!/bin/bash
# STARalign 0.5.19
# Generated by dx-app-wizard.

main() {

    dx download "$fq1" -o ${pair_id}.trim.R1.fastq.gz
    dx download "$fq2" -o ${pair_id}.trim.R2.fastq.gz
    dx download "$reference" -o rnaref.tar.gz

    tar xvfz rnaref.tar.gz
    umiopt=""
    if [[ ${mdup} == 'fgbio_umi' ]]
    then
	umiopt=" -u"
    fi
    
    docker run -v ${PWD}:/data docker.io/goalconsortium/ralign:0.5.19 /seqprg/genomeseer/process_scripts/alignment/rnaseqalign.sh -a ${aligner} -p ${pair_id} -r rnaref -x ${pair_id}.trim.R1.fastq.gz -y ${pair_id}.trim.R2.fastq.gz $umiopt

    docker run -v ${PWD}:/data docker.io/goalconsortium/ralign:0.5.19 /seqprg/genomeseer/process_scripts/alignment/starfusion.sh -p ${pair_id} -r rnaref -a ${pair_id}.trim.R1.fastq.gz -b ${pair_id}.trim.R2.fastq.gz -f

    bam=$(dx upload ${pair_id}.bam --brief)
    bai=$(dx upload ${pair_id}.bam.bai --brief)
    alignstats=$(dx upload ${pair_id}.alignerout.txt --brief)
    starfusion=$(dx upload ${pair_id}.starfusion.txt --brief)

    dx-jobutil-add-output bam "$bam" --class=file
    dx-jobutil-add-output bai "$bai" --class=file
    dx-jobutil-add-output alignstats "$alignstats" --class=file
    dx-jobutil-add-output starfusion "$starfusion" --class=file
}
