#!/bin/bash
# Variant_Calling 0.5.25
# Generated by dx-app-wizard.

main() {

    dx download "$Tumor_BAM" -o ${pair_id}.tumor.bam
    dx download "$reference" -o ref.tar.gz
    mkdir dnaref
    tar xvfz ref.tar.gz --strip-components=1 -C dnaref

    if [ -n "$panel" ]
    then
        dx download "$panel" -o panel.tar.gz
        tar xvfz panel.tar.gz
	capturebed=targetpanel.bed
	pon=mutect2.pon.vcf.gz
    fi

    if [ -n "$Normal_BAM" ]
    then
        dx download "$Normal_BAM" -o ${pair_id}.normal.bam
    fi

    ponopt=''
    if [ -f "$pon" ]
    then
        ponopt='-q $ponfile'
    fi

    normopt=''
    if [[ -n "$Normal_BAM" ]]
    then
        normopt=" -n ${pair_id}.normal.bam"
    fi

    docker run -v ${PWD}:/data docker.io/goalconsortium/variantcalling:0.5.25 bash /seqprg/genomeseer/process_scripts/alignment/indexbams.sh

    if [[ "${algo}" == "fb" ]] || [[ "${algo}" == "platypus" ]]
    then
        docker run -v ${PWD}:/data docker.io/goalconsortium/variantcalling:0.5.25 bash /seqprg/genomeseer/process_scripts/variants/germline_vc.sh -r dnaref -p ${pair_id} -a ${algo} -b $capturebed
        docker run -v ${PWD}:/data docker.io/goalconsortium/variantcalling:0.5.25 bash /seqprg/genomeseer/process_scripts/variants/uni_norm_annot.sh -g 'GRCh38.92' -r dnaref -p ${pair_id}.${algo} -v ${pair_id}.${algo}.vcf.gz

    elif [[ "${algo}" == "strelka2" ]] || [[ "${algo}" == "mutect" ]] || [[ "${algo}" == "shimmer" ]]
    then
        if [[ -z "$Normal_BAM" ]]
        then
            docker run -v ${PWD}:/data docker.io/goalconsortium/variantcalling:0.5.25 bash /seqprg/genomeseer/process_scripts/variants/germline_vc.sh -r dnaref -p ${pair_id} -a ${algo} -b $capturebed ${ponopt}
        else
            docker run -v ${PWD}:/data docker.io/goalconsortium/variantcalling:0.5.25 bash /seqprg/genomeseer/process_scripts/variants/somatic_vc.sh -r dnaref -p ${pair_id} -n ${pair_id}.normal.bam -t ${pair_id}.tumor.bam -a ${algo} -b $capturebed ${ponopt}
        fi
        docker run -v ${PWD}:/data docker.io/goalconsortium/variantcalling:0.5.25 bash /seqprg/genomeseer/process_scripts/variants/uni_norm_annot.sh -g 'GRCh38.92' -r dnaref -p ${pair_id}.${algo} -v ${pair_id}.${algo}.vcf.gz

    else
        echo "Incorrect algorithm selection. Please select 1 of the following algorithms: fb, platypus, strelka2, mutect, shimmer"
    fi
    
    vcf=$(dx upload ${pair_id}.${algo}.vcf.gz --brief)
    ori=$(dx upload ${pair_id}.${algo}.ori.vcf.gz --brief)
    
    dx-jobutil-add-output vcf "$vcf" --class=file
    dx-jobutil-add-output ori "$ori" --class=file
}
