#!/bin/bash
# Variant_Calling 0.5.26
# Generated by dx-app-wizard.

#echo "dx download "$Tumor_BAM" -o ${pair_id}.tumor.bam
# echo "dx download "$reference" -o ref.tar.gz
echo "mkdir dnaref"
echo "tar xvfz ref.tar.gz --strip-components=1 -C dnaref"

if [ -n "$panel" ]
then
    echo "dx download "$panel" -o panel.tar.gz"
    echo "tar xvfz panel.tar.gz"
    echo "capturebed=targetpanel.bed"
    echo "pon=mutect2.pon.vcf.gz"
fi

if [ -n "$Normal_BAM" ]
then
    echo "dx download "$Normal_BAM" -o ${pair_id}.normal.bam"
fi

ponopt=''
if [ -f "$pon" ]
then
    ponopt='-q $ponfile'
fi

normopt=''
if [[ -n "$Normal_BAM" ]]
then
    normopt=" -n ${pair_id}.normal.bam"
    fi

echo "docker run -v ${PWD}:/data docker.io/goalconsortium/variantcalling:0.5.26 bash /seqprg/genomeseer/process_scripts/alignment/indexbams.sh"

vcfout=''
vcfori=''
outfile=''
echo $algo

if [[ "${algo}" == "mutect" ]]
then
    if [[ -z "$Normal_BAM" ]]
    then
        echo "docker run -v ${PWD}:/data docker.io/goalconsortium/variantcalling:0.5.26 bash /seqprg/genomeseer/process_scripts/variants/germline_vc.sh -r dnaref -p ${pair_id} -a ${algo} -b $capturebed ${ponopt}"
    else
        echo "docker run -v ${PWD}:/data docker.io/goalconsortium/variantcalling:0.5.26 bash /seqprg/genomeseer/process_scripts/variants/somatic_vc.sh -r dnaref -p ${pair_id} -n ${pair_id}.normal.bam -t ${pair_id}.tumor.bam -a ${algo} -b $capturebed ${ponopt}"
    fi
    vcfout+=" ${pair_id}.${algo}.vcf.gz"
    vcfori+=" ${pair_id}.${algo}.ori.vcf.gz"
    outfile=".mutect"
else
    for a in $algo
    do
	if [[ "${a}" == "fb" ]] || [[ "${a}" == "platypus" ]]
	then
	    echo "docker run -v ${PWD}:/data docker.io/goalconsortium/variantcalling:0.5.26 bash /seqprg/genomeseer/process_scripts/variants/germline_vc.sh -r dnaref -p ${pair_id} -a ${a} -b $capturebed"
	    echo "docker run -v ${PWD}:/data docker.io/goalconsortium/variantcalling:0.5.26 bash /seqprg/genomeseer/process_scripts/variants/uni_norm_annot.sh -g 'GRCh38.92' -r dnaref -p ${pair_id}.${a} -v ${pair_id}.${a}.vcf.gz"
	    
	elif [[ "${a}" == "strelka2" ]] || [[ "${a}" == "shimmer" ]]
	then
	    if [[ -z "$Normal_BAM" ]]
	    then
		echo "docker run -v ${PWD}:/data docker.io/goalconsortium/variantcalling:0.5.26 bash /seqprg/genomeseer/process_scripts/variants/germline_vc.sh -r dnaref -p ${pair_id} -a ${a} -b $capturebed ${ponopt}"
	    else
		echo "docker run -v ${PWD}:/data docker.io/goalconsortium/variantcalling:0.5.26 bash /seqprg/genomeseer/process_scripts/variants/somatic_vc.sh -r dnaref -p ${pair_id} -n ${pair_id}.normal.bam -t ${pair_id}.tumor.bam -a ${a} -b $capturebed ${ponopt}"
		fi
	    echo "docker run -v ${PWD}:/data docker.io/goalconsortium/variantcalling:0.5.26 bash /seqprg/genomeseer/process_scripts/variants/uni_norm_annot.sh -g 'GRCh38.92' -r dnaref -p ${pair_id}.${a} -v ${pair_id}.${a}.vcf.gz"
	else
	    echo "Incorrect algorithm selection. Please select 1 of the following algorithms: fb, platypus, strelka2, mutect, shimmer"
	fi
	vcfout+=" ${pair_id}.${a}.vcf.gz"
	vcfori+=" ${pair_id}.${a}.ori.vcf.gz"
	outfile+=".${a}"
    done
fi

echo "tar cf ${pair_id}${outfile}.vcfout.tar $vcfout"
echo "tar cf ${pair_id}${outfile}.ori.tar $vcfori"
echo "gzip ${pair_id}${outfile}.vcfout.tar"
echo "gzip ${pair_id}${outfile}.ori.tar"


#vcf=$(dx upload ${pair_id}.${outfile}.vcfout.tar.gz --brief)
#ori=$(dx upload ${pair_id}.${outfile}.ori.tar.gz --brief)

#dx-jobutil-add-output vcf "$vcf" --class=file
#dx-jobutil-add-output ori "$ori" --class=file
