#!/bin/bash
# STARalign 0.0.1
# Generated by dx-app-wizard.

main() {

    dx download "$fq1" -o trim.R1.fastq.gz
    dx download "$fq2" -o trim.R2.fastq.gz
    dx download "$reference" -o dnaref.tar.gz

    tar xvfz dnaref.tar.gz

    if [[ ${mdup} == 'fgbio_umi' ]]
    then
        docker run -v ${PWD}:/data docker.io/goalconsortium/ralign:1.0.0 /usr/local/bin/rnaseqalign.sh -a ${aligner} -p ${pair_id} -r dnaref -x trim.R1.fastq.gz -y trim.R2.fastq.gz -u
    else
        docker run -v ${PWD}:/data docker.io/goalconsortium/ralign:1.0.0 /usr/local/bin/rnaseqalign.sh -a ${aligner} -p ${pair_id} -r dnaref -x trim.R1.fastq.gz -y trim.R2.fastq.gz
    fi
    docker run -v ${PWD}:/data docker.io/goalconsortium/ralign:1.0.0 /usr/local/bin/starfusion.sh -p ${pair_id} -r dnaref -a trim.R1.fastq.gz -b trim.R2.fastq.gz -m trinity -f

    rawbam=$(dx upload ${pair_id}.bam --brief)
    rawbai=$(dx upload ${pair_id}.bam.bai --brief)
    alignstats=$(dx upload ${pair_id}.alignerout.txt --brief)
    starfusion=$(dx upload ${pair_id}.starfusion.txt --brief)

    dx-jobutil-add-output rawbam "$rawbam" --class=file
    dx-jobutil-add-output rawbai "$rawbai" --class=file
    dx-jobutil-add-output alignstats "$alignstats" --class=file
    dx-jobutil-add-output starfusion "$starfusion" --class=file
}
