#!/bin/bash
# QC_Stats 0.5.19
# Generated by dx-app-wizard.

main() {

    dx download "$Raw_BAM" -o ${pair_id}.bam
    dx download "$Raw_BAI" -o ${pair_id}.bam.bai
    dx download "$reference" -o dnaref.tar.gz
    dx download "$panel" -o panel.tar.gz
    dx download "$trimstat" -o ${pair_id}.trimreport.txt

    tar xvfz dnaref.tar.gz
    tar xvfz panel.tar.gz

    USER=$(dx whoami)

    docker run -v ${PWD}:/data docker.io/goalconsortium/vcfannot:0.5.19 bash /seqprg/genomeseer/process_scripts/alignment/bamqc.sh -c targetpanel.bed -n dna -r dnaref -b ${pair_id}.bam -p ${pair_id}

    flagstats=$(dx upload ${pair_id}.flagstat.txt --brief)
    covhist=$(dx upload ${pair_id}.covhist.txt --brief)
    genomecov=$(dx upload ${pair_id}.genomecov.txt --brief)
    otflagstats=$(dx upload ${pair_id}.ontarget.flagstat.txt --brief)
    mapqualcov=$(dx upload ${pair_id}.mapqualcov.txt --brief)
    seqstats=$(dx upload ${pair_id}.sequence.stats.txt --brief)

    dx-jobutil-add-output flagstats "$flagstats" --class=file
    dx-jobutil-add-output covhist "$covhist" --class=file
    dx-jobutil-add-output genomecov "$genomecov" --class=file
    dx-jobutil-add-output otflagstats "$otflagstats" --class=file
    dx-jobutil-add-output mapqualcov "$mapqualcov" --class=file
    dx-jobutil-add-output seqstats "$seqstats" --class=file
}
