#!/bin/bash
# GATK 0.5.25
# Generated by dx-app-wizard.

main() {

    dx download "$Consensus_BAM" -o ${pair_id}.consensus.bam
    dx download "$Consensus_BAI" -o ${pair_id}.consensus.bam.bai
    dx download "$reference" -o ref.tar.gz

    mkdir dnaref
    tar xvfz ref.tar.gz --strip-components=1 -C dnaref

    docker run -v ${PWD}:/data docker.io/goalconsortium/variantcalling:0.5.25 bash /seqprg/genomeseer/process_scripts/variants/gatkrunner.sh -a gatkbam -b ${pair_id}.consensus.bam -r dnaref -p ${pair_id}

    finalbam=$(dx upload ${pair_id}.final.bam --brief)
    finalindex=$(dx upload ${pair_id}.final.bam.bai --brief)

    dx-jobutil-add-output finalbam "$finalbam" --class=file
    dx-jobutil-add-output finalindex "$finalindex" --class=file
}
