#!/bin/bash
# align_markdups 0.5.32
# Generated by dx-app-wizard.

main() {

    dx download "$fq1" -o ${pair_id}.trim.R1.fastq.gz
    dx download "$fq2" -o ${pair_id}.trim.R2.fastq.gz
    dx download "$humanref" -o humanref.tar.gz
    dx download "$panel" -o panel.tar.gz
    dx download "$trimstat" -o ${pair_id}.trimreport.txt

    mkdir humanref
    docker run -v ${PWD}:/data docker.io/goalconsortium/alignment:0.5.32 tar -I pigz -xvf humanref.tar.gz --strip-components=1 -C humanref
    docker run -v ${PWD}:/data docker.io/goalconsortium/alignment:0.5.32 tar -I pigz -xvf panel.tar.gz
    
    alignopt=''
    if [[ ${mdup} == 'fgbio_umi' ]]
    then
	alignopt=" -u"
    fi
    
    docker run -v ${PWD}:/data docker.io/goalconsortium/alignment:0.5.32 bash /seqprg/school/process_scripts/alignment/dnaseqalign.sh -r humanref -p ${pair_id} -x ${pair_id}.trim.R1.fastq.gz -y ${pair_id}.trim.R2.fastq.gz $alignopt
    docker run -v ${PWD}:/data docker.io/goalconsortium/alignment:0.5.32 bash /seqprg/school/process_scripts/alignment/markdups.sh -a ${mdup} -b ${pair_id}.bam -p ${pair_id} -r humanref
    
    mv ${pair_id}.dedup.bam ${pair_id}.consensus.bam
    mv ${pair_id}.dedup.bam.bai ${pair_id}.consensus.bam.bai
    USER=$(dx whoami)

    docker run -v ${PWD}:/data docker.io/goalconsortium/vcfannot:0.5.32 bash /seqprg/school/process_scripts/alignment/bamqc.sh -c targetpanel.bed -n dna -r ./ -b ${pair_id}.bam -p ${pair_id} -u $USER
    tar -czvf ${pair_id}.sequence.stats.tar.gz ${pair_id}.flagstat.txt ${pair_id}.covhist.txt ${pair_id}.genomecov.txt ${pair_id}.ontarget.flagstat.txt  ${pair_id}.mapqualcov.txt ${pair_id}.sequence.stats.txt 
    

    if [[ -n $virusref ]]
    then
	dx download "$virusref" -o virusref.tar.gz
	mkdir virusref
	tar xvfz virusref.tar.gz --strip-components=1 -C virusref

	docker run -v ${PWD}:/data docker.io/goalconsortium/alignment:0.5.32 bash /seqprg/school/process_scripts/alignment/virusalign.sh -b ${pair_id}.bam -p ${pair_id} -r virusref -f
	vseqstat=$(dx upload ${pair_id}.viral.seqstats.txt --brief)
	dx-jobutil-add-output vseqstat "$vseqstat" --class=file
    fi

    rawbam=$(dx upload ${pair_id}.bam --brief)
    conbam=$(dx upload ${pair_id}.consensus.bam --brief)
    rawbai=$(dx upload ${pair_id}.bam.bai --brief)
    conbai=$(dx upload ${pair_id}.consensus.bam.bai --brief)
    seqstats=$(dx upload ${pair_id}.sequence.stats.tar.gz --brief)
    dx-jobutil-add-output seqstats "$seqstats" --class=file
    dx-jobutil-add-output rawbam "$rawbam" --class=file
    dx-jobutil-add-output conbam "$conbam" --class=file
    dx-jobutil-add-output rawbai "$rawbai" --class=file
    dx-jobutil-add-output conbai "$conbai" --class=file

}
