#!/bin/bash
# QC_BAM 0.0.1
# Generated by dx-app-wizard.
#
# Basic execution pattern: Your app will run on a single machine from
# beginning to end.
#
# Your job's input variables (if any) will be loaded as environment
# variables before this script runs.  Any array inputs will be loaded
# as bash arrays.
#
# Any code outside of main() (or any entry point you may add) is
# ALWAYS executed, followed by running the entry point itself.
#
# See https://wiki.dnanexus.com/Developer-Portal for tutorials on how
# to modify this file.

main() {

    dx download "$Consensus_BAM" -o consensus.bam
    dx download "$reference_files" -o reference.tar.gz
    dx download "$capture_bed" -o capture.bed

    tar xvfz reference.tar.gz
    gunzip reference/genome.fa.gz

    docker run -v ${PWD}:/data docker.io/goalconsortium/alignment:v1 bash /usr/local/bin/bam2tdf.sh -r reference -b consensus.bam -p ${pair_id}.uniq
    docker run -v ${PWD}:/data docker.io/goalconsortium/alignment:v1 bash /usr/local/bin/bamqc.sh -c capture.bed -n dna -r reference -b consensus.bam -p ${pair_id} -s 1
    mv ${pair_id}.genomecov.txt ${pair_id}.dedupcov.txt
    mv ${pair_id}.covhist.txt ${pair_id}.covuniqhist.txt
    mv ${pair_id}.hist.txt ${pair_id}.uniqhist.txt
    mv ${pair_id}_lowcoverage.txt ${pair_id}_lowcoverageuniq.txt
    mv ${pair_id}_exoncoverage.txt ${pair_id}_exoncoverageuniq.txt

    umihist=$(dx upload ${pair_id}.uniqhist.txt --brief)
    dedupcov=$(dx upload ${pair_id}.dedupcov.txt --brief)
    covuniqhist=$(dx upload ${pair_id}.covuniqhist.txt --brief)
    lowcovuniqstat=$(dx upload ${pair_id}_lowcoverageuniq.txt --brief)
    exoncovuniqstat=$(dx upload ${pair_id}_exoncoverageuniq.txt --brief)
    uniqtdf=$(dx upload ${pair_id}.uniq.tdf --brief)

    dx-jobutil-add-output umihist "$umihist" --class=file
    dx-jobutil-add-output dedupcov "$dedupcov" --class=file
    dx-jobutil-add-output covuniqhist "$covuniqhist" --class=file
    dx-jobutil-add-output lowcovuniqstat "$lowcovuniqstat" --class=file
    dx-jobutil-add-output exoncovuniqstat "$exoncovuniqstat" --class=file
    dx-jobutil-add-output uniqtdf "$uniqtdf" --class=file
}
