#!/bin/bash
# Structural_Variants 0.5.36
# Generated by dx-app-wizard.

main() {

    dx download "$Tumor_BAM" -o ${pair_id}.tumor.bam
    dx download "$reference" -o ref.tar.gz
    
    normopt=''
    if [ -n "$Normal_BAM" ]
    then
        dx download "$Normal_BAM" -o ${pair_id}.normal.bam
	normopt=" -n ${pair_id}.normal.bam"
    fi

    mkdir dnaref
    docker run -v ${PWD}:/data docker.io/goalconsortium/vcfannot:0.5.36 tar -I pigz -xvf ref.tar.gz --strip-components=1 -C dnaref

    if [ -n "$panel" ]
    then
        dx download "$panel" -o panel.tar.gz
         docker run -v ${PWD}:/data docker.io/goalconsortium/vcfannot:0.5.36 tar -I pigz -xvf panel.tar.gz
    fi

    docker run -v ${PWD}:/data docker.io/goalconsortium/vcfannot:0.5.36 bash /seqprg/school/process_scripts/alignment/indexbams.sh

    if [[ -n "$Normal_BAM" ]]
    then
	
	docker run -v ${PWD}:/data docker.io/goalconsortium/vcfannot:0.5.36 bash /seqprg/school/process_scripts/variants/checkmate.sh -r dnaref -p ${pair_id} -c dnaref/NGSCheckMate.bed -f
	echo -e "TumorFILE\t${Tumor_BAM}" >> ${pair_id}.sequence.stats.txt
	echo -e "NormalFILE\t${Normal_BAM}" >> ${pair_id}.sequence.stats.txt
	
        matched=$(dx upload ${pair_id}_matched.txt --brief)
        all=$(dx upload ${pair_id}_all.txt --brief)
        seqstats=$(dx upload ${pair_id}.sequence.stats.txt --brief)
	
        dx-jobutil-add-output matched "$matched" --class=file
        dx-jobutil-add-output all "$all" --class=file
        dx-jobutil-add-output seqstats "$seqstats" --class=file
    fi
    docker run -v ${PWD}:/data docker.io/goalconsortium/vcfannot:0.5.36 bash /seqprg/school/process_scripts/variants/msisensor.sh -r dnaref -p ${pair_id} -b ${pair_id}.tumor.bam -c targetpanel.bed $normopt
    msiout=$(dx upload ${pair_id}.msi --brief)
    dx-jobutil-add-output msiout "$msiout" --class=file


}
