#!/bin/bash
# Structural_Variants 0.5.25
# Generated by dx-app-wizard.

main() {

    dx download "$Tumor_BAM" -o ${pair_id}.tumor.bam
    dx download "$reference" -o ref.tar.gz

    mkdir dnaref
    tar xvfz ref.tar.gz --strip-components=1 -C dnaref

    if [ -n "$panel" ]
    then
        dx download "$panel" -o panel.tar.gz
        tar xvfz panel.tar.gz
    fi

    if [ -n "$Normal_BAM" ]
    then
        dx download "$Normal_BAM" -o ${pair_id}.normal.bam
    fi

    docker run -v ${PWD}:/data docker.io/goalconsortium/variantcalling:0.5.25 bash /seqprg/genomeseer/process_scripts/alignment/indexbams.sh

    normopt=''
    if [[ -n "$Normal_BAM" ]]
    then
	normopt=" -n ${pair_id}.normal.bam"
    fi

    if [[ "${algo}" == "pindel" ]]
    then
	 docker run -v ${PWD}:/data docker.io/goalconsortium/structuralvariant:0.5.25 -r dnaref -p $pair_id -l dnaref/itd_genes.bed -a pindel -f

         vcf=$(dx upload ${pair_id}.${algo}.vcf.gz --brief)
         tdvcf=$(dx upload ${pair_id}.${algo}_tandemdup.vcf.gz --brief)
         genefusion=$(dx upload ${pair_id}.${algo}.genefusion.txt --brief)
	 
         dx-jobutil-add-output vcf "$vcf" --class=file
         dx-jobutil-add-output tdvcf "$tdvcf" --class=file
         dx-jobutil-add-output genefusion "$genefusion" --class=file
	 
    elif [[ "${algo}" == "delly" ]] || [[ "${algo}" == "svaba" ]]
    then
        docker run -v ${PWD}:/data docker.io/goalconsortium/structuralvariant:0.5.25 -r dnaref -b ${pair_id}.tumor.bam -p ${pair_id} -a ${algo} $normopt -f
	
        vcf=$(dx upload ${pair_id}.${algo}.vcf.gz --brief)
        genefusion=$(dx upload ${pair_id}.${algo}.genefusion.txt --brief)
	
        dx-jobutil-add-output vcf "$vcf" --class=file
        dx-jobutil-add-output genefusion "$genefusion" --class=file
	
    elif [[ "${algo}" == "itdseek" ]]
    then
	docker run -v ${PWD}:/data docker.io/goalconsortium/vcfannot:0.5.25 bash  /seqprg/genomeseer/process_scripts/variants/svcalling.sh -r dnaref -b ${pair_id}.tumor.bam -p ${pair_id} -a ${algo} -l dnaref/itd_genes.bed -f
	vcf=$(dx upload ${pair_id}.itdseek_tandemdup.vcf.gz --brief)
	dx-jobutil-add-output vcf "$vcf" --class=file

    else
        echo "Incorrect algorithm selection. Please select 1 of the following algorithms: pindel, delly, svaba"
    fi
}
