#!/bin/bash
# Structural_Variants 0.5.33
# Generated by dx-app-wizard.

#dx download "$Tumor_BAM" -o ${pair_id}.tumor.bam
#dx download "$reference" -o ref.tar.gz

#mkdir dnaref
#tar xvfz ref.tar.gz --strip-components=1 -C dnaref

if [ -n "$panel" ]
then
    dx download "$panel" -o panel.tar.gz
    mkdir -p panel
    tar xvfz panel.tar.gz -C panel/
fi

if [ -n "$Normal_BAM" ]
then
    dx download "$Normal_BAM" -o ${pair_id}.normal.bam
fi

echo "docker run -v ${PWD}:/data docker.io/goalconsortium/structuralvariant:0.5.33 bash /seqprg/school/process_scripts/alignment/indexbams.sh"
    
normopt=''
if [[ -n "$Normal_BAM" ]]
then
    normopt=" -n ${pair_id}.normal.bam"
fi

vcfout=''
vcfsv=''
gffile=''
outfile=''

echo "$algo"

for a in $algo
do
    echo "Starting ${a}"
    outfile+=".${a}"
    
    if [[ "${a}" == "pindel" ]]
    then
	echo "docker run -v ${PWD}:/data docker.io/goalconsortium/structuralvariant:0.5.33 bash /seqprg/school/process_scripts/variants/svcalling.sh -r dnaref -p $pair_id -l dnaref/itd_genes.bed -a pindel -f"
	vcfout+=" ${pair_id}.${a}.vcf.gz"
	vcfsv+=" ${pair_id}.${a}_tandemdup.vcf.gz"
	gffile+=" ${pair_id}.${a}.genefusion.txt"
    elif [[ "${a}" == "delly" ]] || [[ "${a}" == "svaba" ]]
    then
        echo "docker run -v ${PWD}:/data docker.io/goalconsortium/structuralvariant:0.5.33 bash /seqprg/school/process_scripts/variants/svcalling.sh -r dnaref -b ${pair_id}.tumor.bam -p ${pair_id} -a ${a} $normopt -f"
        vcfsv+=" ${pair_id}.${a}.vcf.gz"
	gffile+=" ${pair_id}.${a}.genefusion.txt"
    elif [[ "${a}" == "itdseek" ]]
    then
	echo "docker run -v ${PWD}:/data docker.io/goalconsortium/structuralvariant:0.5.33 bash /seqprg/school/process_scripts/variants/svcalling.sh -r dnaref -b ${pair_id}.tumor.bam -p ${pair_id} -a ${a} -l dnaref/itd_genes.bed -f"
	vcfsv+=" ${pair_id}.${a}_tandemdup.vcf.gz"
    elif [[ "${a}" == "cnvkit" ]]
    then
	echo "docker run -v ${PWD}:/data docker.io/goalconsortium/structuralvariant:0.5.33 bash /seqprg/school/process_scripts/variants/cnvkit.sh -r dnaref -b ${pair_id}.tumor.bam -p ${pair_id} -d panel"
	echo "tar -czvf ${pair_id}.cnvout.tar.gz ${pair_id}.answerplot* ${pair_id}.*txt ${pair_id}.call.cns ${pair_id}.cns ${pair_id}.cnr"
    else
        echo "Incorrect algorithm selection. Please select 1 of the following algorithms: pindel, delly, svaba,cnvkit"
    fi
done

#tar -czvf ${pair_id}${outfile}.vcf.tar.gz vcfout
#tar -czvf ${pair_id}${outfile}.sv.tar.gz vcfsv
#tar -czvf ${pair_id}${outfile}.gf.tar.gz  gffile


#vcf=$(dx upload ${pair_id}${outfile}.vcf.tar.gz --brief)
#svvcf=$(dx upload ${pair_id}${outfile}.sv.tar.gz --brief)
#genefusion=$(dx upload ${pair_id}${outfile}.gf.tar.gz --brief)
#cnvout=$(dx upload ${pair_id}${outfile}.gf.tar.gz --brief)

#dx-jobutil-add-output cnvout "$cnvout" --class=file
#dx-jobutil-add-output vcf "$vcf" --class=file
#dx-jobutil-add-output tdvcf "$svvcf" --class=file
#dx-jobutil-add-output genefusion "$genefusion" --class=file
