#!/bin/bash
# bamct_alignQC 0.5.27
# Generated by dx-app-wizard.

main() {

    dx download "$Aligned_BAM" -o ${pair_id}.bam
    dx download "$Align_Stats" -o ${pair_id}.alignerout.txt

    USER=$(dx whoami)
    if [[ -n "${bamct}" ]]
    then
        dx download "$bamct" -o ref.tar.gz

        mkdir rnaref
        tar xvfz ref.tar.gz --strip-components=1 -C rnaref

        docker run -v ${PWD}:/data docker.io/goalconsortium/vcfannot:0.5.27 samtools index ${pair_id}.bam
        docker run -v ${PWD}:/data docker.io/goalconsortium/vcfannot:0.5.27 bam-readcount -w 0 -q 0 -b 25 -f rnaref/genome.fa ${pair_id}.bam > ${pair_id}.bamreadcount.txt

        bamreadct=$(dx upload ${pair_id}.bamreadcount.txt --brief)

        dx-jobutil-add-output bamreadct "$bamreadct" --class=file
    fi
    
    docker run -v ${PWD}:/data docker.io/goalconsortium/vcfannot:0.5.27 bash /seqprg/genomeseer/process_scripts/alignment/bamqc.sh -p ${pair_id} -b ${pair_id}.bam -n rna 

    fastqczip=$(dx upload ${pair_id}_fastqc.zip --brief)
    fastqchtml=$(dx upload ${pair_id}_fastqc.html --brief)
    seqstats=$(dx upload ${pair_id}.sequence.stats.txt --brief)

    dx-jobutil-add-output fastqczip "$fastqczip" --class=file
    dx-jobutil-add-output fastqchtml "$fastqchtml" --class=file
    dx-jobutil-add-output seqstats "$seqstats" --class=file
}
